lib/lambda_ethereum_consensus/execution/execution_client.ex

```diff
@@ -0,0 +1,109 @@
+defmodule LambdaEthereumConsensus.Execution.ExecutionClient do
+  @moduledoc """
+  Execution Layer Engine API methods
+  """
+
+  use GenServer
+  require Logger
+
+  alias LambdaEthereumConsensus.Execution.Auth
+  alias LambdaEthereumConsensus.Execution.EngineApi
+
+  @type state :: %{
+          jwt_secret: binary()
+        }
+
+  ##########################
+  ### Public API
+  ##########################
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  def engine_exchange_capabilities do
+    GenServer.call(__MODULE__, :engine_exchange_capabilities)
+  end
+
+  def forkchoice_updated(forkchoice_state, payload_attributes) do
+    GenServer.call(__MODULE__, {:forkchoice_updated, forkchoice_state, payload_attributes})
+  end
+
+  @spec verify_and_notify_new_payload(any()) :: {:ok, true}
+  def verify_and_notify_new_payload(_execution_payload) do
+    # TODO: call engine api
+    {:ok, true}
+  end
+
+  ##########################
+  ### GenServer Callbacks
+  ##########################
+
+  @impl GenServer
+  @spec init(any) :: {:ok, state()} | {:error, any}
+  def init([engine_endpoint, jwt_path]) do
+    if engine_endpoint do
```

Comment:
should we do these checks in a different module for all cli options instead of the module that uses them?

Reply:
I think we should. That way, we can validate the options all in one place. We also want to enable or disable some of the supervision tree depending on the CLI flags.

Reply:
done